rdkit/Chem/Fingerprints/DbFpSupplier.py

```diff
@@ -105,17 +105,13 @@ def NextItem(self):
           NOTE: this has side effects
 
         """
-        try:
-            d = next(self._dataIter)
-        except StopIteration:
-            d = None
+        d = next(self._dataIter, None)
+        # return self._BuildFp(next(self._dataIter, None))
         if d is not None:
-            newFp = self._BuildFp(d)
-        else:
-            newFp = None
-        return newFp
-
-
+            return self._BuildFp(d)
+        return None 
```

Comment:
That is nicer. Remove the commented line.

Reply:
OK, I got your idea! I would fix it in the next patch. Thank you for your review.